tests/sentry/incidents/serializers/test_workflow_engine_detector.py

```diff
@@ -0,0 +1,222 @@
+from typing import Any
+from unittest.mock import patch
+
+from sentry.api.serializers import serialize
+from sentry.incidents.endpoints.serializers.workflow_engine_detector import (
+    WorkflowEngineDetectorSerializer,
+)
+from sentry.incidents.models.alert_rule import (
+    AlertRuleStatus,
+    AlertRuleThresholdType,
+    AlertRuleTriggerAction,
+)
+from sentry.incidents.models.incident import IncidentTrigger, TriggerStatus
+from sentry.models.groupopenperiod import GroupOpenPeriod
+from sentry.testutils.cases import TestCase
+from sentry.testutils.helpers.datetime import freeze_time
+from sentry.types.group import PriorityLevel
+from sentry.workflow_engine.migration_helpers.alert_rule import (
+    migrate_alert_rule,
+    migrate_metric_action,
+    migrate_metric_data_conditions,
+    migrate_resolve_threshold_data_condition,
+)
+from sentry.workflow_engine.models import ActionGroupStatus
+
+
+@freeze_time("2024-12-11 03:21:34")
+class TestDetectorSerializer(TestCase):
+    def setUp(self) -> None:
+        self.alert_rule = self.create_alert_rule()
+        self.critical_trigger = self.create_alert_rule_trigger(
+            alert_rule=self.alert_rule, label="critical"
+        )
+        self.critical_trigger_action = self.create_alert_rule_trigger_action(
+            alert_rule_trigger=self.critical_trigger
+        )
+        self.warning_trigger = self.create_alert_rule_trigger(
+            alert_rule=self.alert_rule, label="warning"
+        )
+        self.warning_trigger_action = self.create_alert_rule_trigger_action(
+            alert_rule_trigger=self.warning_trigger
+        )
+        _, _, _, self.detector, _, _, _, _ = migrate_alert_rule(self.alert_rule)
+        self.critical_detector_trigger, _ = migrate_metric_data_conditions(self.critical_trigger)
+        self.warning_detector_trigger, _ = migrate_metric_data_conditions(self.warning_trigger)
+
+        self.critical_action, _, _ = migrate_metric_action(self.critical_trigger_action)
+        self.warning_action, _, _ = migrate_metric_action(self.warning_trigger_action)
+        self.resolve_trigger_data_condition = migrate_resolve_threshold_data_condition(
+            self.alert_rule
+        )
+        self.expected_critical_action = [
```

Comment:
I want to go back and make a shared base class for these after this is merged, it's already a large diff so I didn't wanna do it now.